runtime/src/iree/hal/drivers/cuda/cuda_driver.c

```diff
@@ -385,6 +385,16 @@ static iree_status_t iree_hal_cuda_driver_create_device_by_index(
   // Ensure CUDA is initialized before querying it.
   IREE_RETURN_IF_ERROR(iree_hal_cuda_init(driver));
 
+  // Query the number of available CUDA devices.
+  int device_count = 0;
+  CUDA_RETURN_IF_ERROR(&driver->syms, cuDeviceGetCount(&device_count),
+                       "cuDeviceGetCount");
+  if (device_index >= device_count) {
+    return iree_make_status(IREE_STATUS_NOT_FOUND,
+                            "default device %d not found (of %d enumerated)",
```

Comment:
```suggestion
                            "device %d not found (of %d enumerated)",
```
(this isn't default)